eZ/Publish/Core/Persistence/Legacy/Content/Location/Handler.php

```diff
@@ -549,4 +549,12 @@ public function loadAllLocations($offset, $limit)
 
         return $this->locationMapper->createLocationsFromRows($rows);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRootLocation(Location $location)
+    {
+        return $location->depth === 0;
```

Comment:
This should rather be `Location::isRoot() : bool`, Persistence Handlers are for lookups only.

But we usually only add that on API, and in API root might mean something else depending on context _(site root)_, so just inlining this logic in `loadParentLocations ` is also more than ok.

Reply:
> This should rather be `Location::isRoot() : bool`, Persistence Handlers are for lookups only.
> 

@andrerom what happened to not keeping business logic in Value Objects? :stuck_out_tongue_winking_eye: 

But given you're not the first suggesting this, we probably need to rethink current approach // cc @adamwojs

Reply:
Well #B is an option then until there is a place to place this kind of logic ;)

Reply:
This is something else @crevillo, if you read the code it's about not trying to load parent if there is none.

Reply:
ok then. i was thinking that the name of the method could confuse someone on the real functiontality of it.

Reply:
> ok then. i was thinking that the name of the method could confuse someone on the real functiontality of it.

yep, yet another reason for approach #B - to inline it ;) 
Indeed we have separate concepts of Root in tree - this one is about low level - SPI\Persistence - layer while the SA aware `tree_root.location_id` refers to another, higher level layer.